experiment/simulation/js/xor.js

```diff
@@ -0,0 +1,156 @@
+            <tr>
+                <td>1</td><td>0</td><td>0</td>
+            </tr>
+            <tr>
+                <td>1</td><td>1</td><td>1</td>
+            </tr>`;
+    }
+    listInput[0].input = initialInput0;
+    listInput[1].input = initialInput1;
+    listInput[2].input = 1 - initialInput0;
+    listInput[3].input = 1 - initialInput1;
+    checkAndUpdate();
+}
+
+export function modifyOutput() {
+    const divOutput0 = document.getElementById("output0");
+    divOutput0.innerHTML = 'Output<br>' + getTruthValue();
+}
+
+export function permutator(inputArr) {
+    const results = [];
+
+    function permute(arr, memo) {
+        let currentCase;
+
+        memo = memo || [];
+
+        for (let i = 0; i < arr.length; i++) {
+            currentCase = arr.splice(i, 1);
+            if (arr.length === 0) {
+                results.push(memo.concat(currentCase));
+            }
+            permute(arr.slice(), memo.concat(currentCase));
+            arr.splice(i, 0, currentCase[0]);
+        }
+
+        return results;
+    }
+
+    return permute(inputArr);
+}
+
+export function checkXor() {
+    const permutatorMap = permutator([0, 1, 2, 3]);
+    let xorCircuitValid = 0;
+    for (let i = 0; i < permutatorMap.length; i++) {
+        for (let j = 0; j < permutatorMap.length; j++) {
+            for (let k = 0; k < permutatorMap.length; k++) {
+                if (connectionMap.has(`input${permutatorMap[i][0]}$pmos${permutatorMap[j][0]}`) && connectionMap.has(`input${permutatorMap[i][1]}$pmos${permutatorMap[j][1]}`) && connectionMap.has(`vdd0$pmos${permutatorMap[j][0]}`) && connectionMap.has(`vdd0$pmos${permutatorMap[j][1]}`) && connectionMap.has(`input${permutatorMap[i][2]}$pmos${permutatorMap[j][2]}`) && connectionMap.has(`input${permutatorMap[i][3]}$pmos${permutatorMap[j][3]}`) && connectionMap.has(`pmos${permutatorMap[j][0]}$pmos${permutatorMap[j][2]}`) && connectionMap.has(`pmos${permutatorMap[j][0]}$pmos${permutatorMap[j][3]}`) && connectionMap.has(`pmos${permutatorMap[j][1]}$pmos${permutatorMap[j][3]}`) && connectionMap.has(`pmos${permutatorMap[j][1]}$pmos${permutatorMap[j][2]}`) && connectionMap.has(`pmos${permutatorMap[j][2]}$output0`) && connectionMap.has(`pmos${permutatorMap[j][3]}$output0`) && (connectionMap.size === 22) && connectionMap.has(`input${permutatorMap[i][0]}$nmos${permutatorMap[k][0]}`) && connectionMap.has(`input${permutatorMap[i][1]}$nmos${permutatorMap[k][2]}`) && connectionMap.has(`ground0$nmos${permutatorMap[k][2]}`) && connectionMap.has(`ground0$nmos${permutatorMap[k][3]}`) && connectionMap.has(`input${permutatorMap[i][2]}$nmos${permutatorMap[k][1]}`) && connectionMap.has(`input${permutatorMap[i][3]}$nmos${permutatorMap[k][3]}`) && connectionMap.has(`nmos${permutatorMap[k][2]}$nmos${permutatorMap[k][0]}`) && connectionMap.has(`nmos${permutatorMap[k][3]}$nmos${permutatorMap[k][1]}`) && connectionMap.has(`nmos${permutatorMap[k][0]}$output0`) && connectionMap.has(`nmos${permutatorMap[k][1]}$output0`)) {
```

Comment:
Can we put this logic inside a separate function instead of putting it inside the if condition?

Reply:
Made a separate function